File: parent335/pinyougou-web-user/src/main/webapp/js/controller/AddressController.js
// 定义控制器:
app.controller("AddressController",function($scope,$http,AddressService,loginService){
    $scope.showName=function(){
        loginService.showName().success(
            function(response){
                $scope.img=response.img;
                $scope.loginName=response.loginName;
                $scope.findAll();
            }
        );
    }
    $scope.findAll = function(){
        // 向后台发送请求:
        AddressService.findAll().success(function(response){
            $scope.list = response;
        });
    }

    $scope.save = function () {
        // 区分是保存还是修改
        var object;
        if($scope.entity.id != null){
            // 更新
            object = AddressService.update($scope.entity);
        }else{
            // 保存
            object = AddressService.insert($scope.entity);
        }
        object.success(function(response) {
            // {flag:true,message:xxx}
            // 判断保存是否成功:
            if (response.flag) {
                // 保存成功
                $scope.findAll();
            } else {
                // 保存失败
                alert(response.message);
            }
        });
    }





    $scope.insert = function(){
        // 向后台发送请求:
        AddressService.insert($scope.entity).success(function(response){
            // 判断保存是否成功:
            if(response.flag){
                // 保存成功
                $scope.findAll();
                 alert(response.message);

            }else{
                // 保存失败
                alert(response.message);
            }
        });
    }

    $scope.delete = function(id){
        // 向后台发送请求:
        if (confirm("确认要删除吗","取消")){
        AddressService.delete(id).success(function(response){
            // 判断删除成功:
            if(response.flag){
                $scope.findAll();
            }else{
                // 保存失败
                alert(response.message);
            }
        });

        }
    }

    $scope.findOne = function(id){
        AddressService.findOne(id).success(function(response){
            $scope.entity = response;
        });
    }

    $scope.clear  =  function () {
        $scope.entity = {};
    }

    $scope.setDefaultAddress = function (id) {
        AddressService.setDefaultAddress(id).success(function(response){
            // 判断设置默认是否成功:
            if(response.flag){
                // 设置成功
                $scope.findAll();
            }else{
                // 设置失败
                alert(response.message);
            }
        });
    }
});

File: parent335/pinyougou-web-user/src/main/java/cn/itcast/core/controller/address/AddressController.java
package cn.itcast.core.controller.address;

import cn.itcast.core.entity.Result;
import cn.itcast.core.pojo.address.Address;
import cn.itcast.core.service.address.AddressService;
import cn.itcast.core.service.address.AreasService;
import cn.itcast.core.service.address.CitiesService;
import cn.itcast.core.service.address.ProvincesService;
import com.alibaba.dubbo.config.annotation.Reference;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/address")
public class AddressController {

    @Reference
    private AddressService addressService;


    /**
     * 根据用户查询他的所有收货地址
     * @return
     */
    @RequestMapping("/findAll.do")
    public List<Address> findAll(){
        String username = SecurityContextHolder.getContext().getAuthentication().getName();
        return  addressService.findAddressList(username);
    }

    /**
     * 根据用户新增一个地址
     * @param address
     * @return
     */
    @RequestMapping("/insert.do")
    public Result insert(@RequestBody Address address){
        String userId = SecurityContextHolder.getContext().getAuthentication().getName();
        address.setUserId(userId);
        List<Address> addressList = addressService.findAddressList(address.getUserId());
        if(addressList == null || addressList.size() == 0){
            address.setIsDefault("1");
        }else {
            address.setIsDefault("0");
        }
        try {
            addressService.insert(address);
            return new Result(true,"添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }
    /**
     * 根据地址id更新
     * @param address
     * @return
     */
    @RequestMapping("/update.do")
    public Result update(@RequestBody Address address){
        try {
            addressService.update(address);
            return new Result(true,"更新成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"更新失败");
        }
    }
    /**
     * 根据地址id更新默认地址
     * @param id
     * @return
     */
    @RequestMapping("/setDefaultAddress.do")
    public Result setDefaultAddress(Long id){
        String userId = SecurityContextHolder.getContext().getAuthentication().getName();
        try {
            addressService.setDefaultAddress(userId,id);
            return new Result(true,"设置默认成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"设置默认失败");
        }
    }

    /**
     * 根据地址id查询当前地址对象
     * @param id
     * @return
     */
    @RequestMapping("/findOne.do")
    public Address findOne(Long id){
        return addressService.findOne(id);
    }

    /**
     * 根据地址id删除
     * @param id
     * @return
     */
    @RequestMapping("/delete.do")
    public Result delete(Long id){
        try {
            addressService.delete(id);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

}
